de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inventory/process/M_InventoryLine_MarkAsCounted.java

```diff
@@ -0,0 +1,76 @@
+package de.metas.inventory.process;
+
+import static org.adempiere.model.InterfaceWrapperHelper.save;
+
+import java.util.Iterator;
+
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.ad.dao.IQueryBuilder;
+import org.adempiere.ad.dao.IQueryFilter;
+import org.adempiere.ad.trx.api.ITrx;
+import org.adempiere.util.Services;
+import org.compiere.model.IQuery;
+import org.compiere.model.I_M_InventoryLine;
+
+import de.metas.process.JavaProcess;
+
+/*
+ * #%L
+ * de.metas.swat.base
+ * %%
+ * Copyright (C) 2018 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+public class M_InventoryLine_MarkAsCounted extends JavaProcess
+{
+
+	@Override
+	protected String doIt() throws Exception
+	{
+		getSelectedInventoryLines()
+				.forEach(inventoryLine -> {
+
+					if (!inventoryLine.isCounted())
+					{
+						inventoryLine.setIsCounted(true);
+						save(inventoryLine);
+					}
+				});
+
+		return MSG_OK;
+	}
+
+	private Iterable<I_M_InventoryLine> getSelectedInventoryLines()
```

Comment:
why not returning List instead of Iterable. Are there so many inventory lines?

Reply:
I don't think there could be so many. And I made it with List in the other process so I am changing this one to List, too.